### PR TITLE
benches: Only use perfcnt on x86

### PR DESCRIPTION
**Pull request by brson**
Fixes ARM build.

Signed-off-by: Brian Anderson <andersrb@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/6639

Problem Summary:

Build is currently broken due to use of the perfcnt crate (and criterion-perf-events), which does not build on arm. These crates are used in a single benchmark.

### What is changed and how it works?

What's Changed:

Only use these crates on x86.

### Release note 

No release note.

### CODE REVIEW

File: tests/benches/coprocessor_executors/mod.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ fn execute<M: criterion::measurement::Measurement + 'static>(c: &mut criterion::
     c.final_summary();
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
```
Comment (kennytm) [inline tests/benches/coprocessor_executors/mod.rs:29]: Please also update the `not` part on line 55

**Review by kennytm — COMMENTED** [review]
Rest LGTM

File: tests/benches/coprocessor_executors/mod.rs (hunk 1 above)
Comment (brson) [reply tests/benches/coprocessor_executors/mod.rs:29]: Done. Good catch.

**Pull request merged**